### PR TITLE
doc: Add ei_ncs library documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds ei_ncs library documentation and two Edge Impulse samples.

Jira: NCSDK-7980, NCSDK-8038

### CODE REVIEW

File: include/ei_ncs.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+The wrapper:
+
+* Buffers input data for the machine learning model.
+* Runs the machine learning model in a separate thread.
+* Provides results via a dedicated callback.
+
+Configuration
+*************
+
+By default, the library is not used in any sample.
```
Comment (pdunaj) [inline include/ei_ncs.rst:26]: remove this line

File: include/ei_ncs.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+   See :ref:`cmake_options` for more information about defining CMake options for command line builds and SEGGER Embedded studio.
+   You can access your API keys under ``Keys`` tab in the Edge Impulse project dashboard.
+
+Enabling the Edge Impulse NCS library
+-------------------------------------
+
+The Edge Impulse NCS library is enabled by :option:`CONFIG_EI_NCS` option.
+The option is enabled by default if :option:`CONFIG_EDGE_IMPULSE` is set.
+
+If you disable the :option:`CONFIG_EI_NCS`, you can directly use the API provided by Edge Impulse C++ library downloaded from Edge Impulse studio.
```
Comment (pdunaj) [inline include/ei_ncs.rst:88]: I would remove this and the following sentence.

**Review by pdunaj — COMMENTED** [review]
I think the content is ok apart from three part I think should be removed.

File: include/ei_ncs.rst (hunk 2 above)
Comment (MarekPieta) [reply include/ei_ncs.rst:88]: done

File: include/ei_ncs.rst (hunk 1 above)
Comment (MarekPieta) [reply include/ei_ncs.rst:26]: done

Comment (pdunaj) [reply include/ei_ncs.rst:26]: I meant the text "By default, the library is not used in any sample." :)

Comment (MarekPieta) [reply include/ei_ncs.rst]: Added: `Complete the following steps to enable and configure the Edge Impulse NCS library.` We need to avoid empty paragraphs.

**Review by pdunaj — APPROVED** [review]

**Review by greg-fer — CHANGES_REQUESTED** [review]
Discussed further changes on Teams.

**Comment by MarekPieta** [comment]
Applied discussed changes

**Review by greg-fer — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**